tests/dataimportcron_test.go

```diff
@@ -0,0 +1,225 @@
+	cdiclientset "kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned"
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	dataImportCronTimeout = 90 * time.Second
+)
+
+var (
+	importsToKeep int32 = 1
+)
+
+var _ = Describe("DataImportCron", func() {
+	const (
+		scheduleEveryMinute = "* * * * *"
+		scheduleOnceAYear   = "0 0 1 1 *"
+	)
+	var (
+		f                   = framework.NewFramework(namespacePrefix)
+		registryPullNode    = cdiv1.RegistryPullNode
+		trustedRegistryURL  = func() string { return fmt.Sprintf(utils.TrustedRegistryURL, f.DockerPrefix) }
+		externalRegistryURL = "docker://quay.io/kubevirt/cirros-container-disk-demo"
+		cron                *cdiv1.DataImportCron
+		err                 error
+	)
+
+	AfterEach(func() {
+		if cron != nil {
+			By("Delete cron")
+			err = DeleteDataImportCron(f.CdiClient, cron.Namespace, cron.Name)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	table.DescribeTable("should", func(schedule string, repeat int, checkGarbageCollection bool) {
+		var url string
+
+		if repeat > 1 || utils.IsOpenshift(f.K8sClient) {
+			url = externalRegistryURL
+		} else {
+			url = trustedRegistryURL()
+			err = utils.AddInsecureRegistry(f.CrClient, url)
+			Expect(err).To(BeNil())
+
+			hasInsecReg, err := utils.HasInsecureRegistry(f.CrClient, url)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hasInsecReg).To(BeTrue())
+			defer utils.RemoveInsecureRegistry(f.CrClient, url)
+		}
+
+		cron = NewDataImportCron("cron-test", "5Gi", schedule, "datasource-test", cdiv1.DataVolumeSourceRegistry{URL: &url, PullMethod: &registryPullNode})
+		By(fmt.Sprintf("Create new DataImportCron %s", url))
+		cron, err = CreateDataImportCronFromDefinition(f.CdiClient, f.Namespace.Name, cron)
+		Expect(err).ToNot(HaveOccurred())
+
+		if schedule == scheduleEveryMinute {
+			var cronjob *v1beta1.CronJob
+			By("Verify cronjob was created")
+			Eventually(func() bool {
+				cronjob, err = f.K8sClient.BatchV1beta1().CronJobs(f.CdiInstallNs).Get(context.TODO(), controller.GetCronJobName(cron), metav1.GetOptions{})
+				return err == nil && len(cronjob.Status.Active) > 0
+			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("Find pod of job %s", cronjob.Status.Active[0].Name))
+			Eventually(func() bool {
+				_, err = utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, cronjob.Status.Active[0].Name, "")
+				return err == nil
+			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+		}
+
+		var lastImportDv, currentImportDv string
+		for i := 0; i < repeat; i++ {
+			if repeat > 1 {
+				// Emulate source update using digests from https://quay.io/repository/kubevirt/cirros-container-disk-demo?tab=tags
+				digest := []string{
+					"sha256:68b44fc891f3fae6703d4b74bcc9b5f24df8d23f12e642805d1420cbe7a4be70",
```

Comment:
Are these shas associated with a tag? I don't think they will get overriden with a different sha in the future, but might want to think about looking up the with a pod at the beginning of the test, and building the digest list from there. I am okay with this for now.

Reply:
To be fixed post-merge.